### PR TITLE
Refactor login to use golang.org/x/oauth2

### PR DESCRIPTION
**Pull request by kitschysynq**
Refactors the login process to use the defacto standard golang oauth2
library. Add an oauth2device package to support the device code flow.
This does not yet update the consumers of the returned token.

### CODE REVIEW

File: oauth2device/oauth2device_test.go [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+func staticHandler(s string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprint(w, s)
+	})
+}
+
+/*
+func (c *Config) DeviceCode(ctx context.Context) (*DeviceCode, error) {
+func (c *Config) Wait(ctx context.Context, code *DeviceCode) (*oauth2.Token, error) {
+*/
```
Comment (justin1121) [inline oauth2device/oauth2device_test.go:274]: Are these needed for something?

**Review by justin1121 — DISMISSED** [review]
LGTM. Just some commented code that I'm not sure needs to exist.

File: oauth2device/oauth2device_test.go (hunk 1 above)
Comment (kitschysynq) [reply oauth2device/oauth2device_test.go:274]: Ah, good catch. I put sigs in the test file as a reference as I'm writing tests and I forgot to remove these.

Comment (justin1121) [reply oauth2device/oauth2device_test.go]: makes sense :grin:

**Review by justin1121 — APPROVED** [review]

**Pull request merged**